network.ts

```diff
@@ -86,7 +86,7 @@ export function isJsonFailure (response: any): response is JsonFailureResponse
  */
 export function isStatusFailure (response: any): response is StatusFailureResponse
 {
-    return response.reason && "invalid_json" === response.reason;
+    return response.reason && "status" === response.reason;
```

Comment:
Interesting that TypeScript didn't complain about this. One might imagine that it must validate this one way or another.